Userland/Libraries/LibWeb/MimeSniff/Resource.cpp

```diff
@@ -0,0 +1,629 @@
+            BytePatternTableRow { "\x3C\x49\x46\x52\x41\x4D\x45\x00"sv.bytes(),
+                "\xFF\xDF\xDF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<H1" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x48\x31\x00"sv.bytes(), "\xFF\xDF\xFF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<DIV" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x44\x49\x56\x00"sv.bytes(), "\xFF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<FONT" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x46\x4F\x4E\x54\x00"sv.bytes(), "\xFF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<TABLE" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x54\x41\x42\x4C\x45\x00"sv.bytes(), "\xFF\xDF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<A" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x41\x00"sv.bytes(), "\xFF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<STYLE" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x53\x54\x59\x4C\x45\x00"sv.bytes(),
+                "\xFF\xDF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<TITLE" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x54\x49\x54\x4C\x45\x00"sv.bytes(),
+                "\xFF\xDF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<B" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x00"sv.bytes(), "\xFF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<BODY" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x4F\x44\x59\x00"sv.bytes(), "\xFF\xDF\xDF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<BR" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x42\x52\x00"sv.bytes(), "\xFF\xDF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
+
+            // The case-insensitive string "<P" followed by a tag-terminating byte.
+            BytePatternTableRow { "\x3C\x50\x00"sv.bytes(), "\xFF\xDF\xFF"sv.bytes(), ignored_whitespace_bytes, text_html_mime_type, true },
```

Comment:
The case-insensitivity doesn't seem to be handled

Reply:
Actually this is something that the pattern matching algorithm considers, see the spec note right below the [pattern mask definition](https://mimesniff.spec.whatwg.org/#pattern-mask) . The byte `\xDF` in each of the pattern masks here represent that the byte is significant but in a ASCII case-insensitive way.

Reply:
> static size_t MAX_SNIFF_SIZE = 1445;
buffer = data.slice(0, min(data.size(), MAX_SNIFF_SIZE));

Hm, I'm having trouble working with `slice()` since it returns a `Span<const unsigned char>` which is not something `ByteBuffer` can use in its constructors. I do like your use of `min()` and decided to remove my if branch in the for loop body and instead used `min(data.size(), MAX_SNIFF_SIZE)` in the for loop statement